### PR TITLE
implement CPLReadLine3L()

### PR DESCRIPTION
**Pull request by landam**
## What does this PR do?

Implements new CPLReadLine3L().

See https://lists.osgeo.org/pipermail/gdal-dev/2018-April/048436.html

## What are related issues/pull requests?

#365 

## Tasklist

 - [ ] Add test case(s)
 - [ ] Review
 - [ ] Adjust for comments
 - [ ] All CI builds and checks have passed

### CODE REVIEW

File: gdal/port/cpl_conv.cpp [hunk 1]
```diff
@@ -651,6 +651,36 @@ const char *CPLReadLineL(VSILFILE *fp) { return CPLReadLine2L(fp, -1, nullptr);
...
+
+/**
+ * Simplified line reading from text file.
+ *
+ * Similar to CPLReadLine(), but reading from a large file API handle.
+ *
+ * @param fp file pointer opened with VSIFOpenL().
+ * @param nMaxCars  maximum number of characters allowed, or -1 for no limit.
+ * @param papszOptions NULL-terminated array of options. Unused for now.
+ * @param nBufLength[out] size of output string
```
Comment (rouault) [inline gdal/port/cpl_conv.cpp]: Should be named pnBufLength to follow naming conventions in GDAL. I'd add in he doc that the pointer must be non-NULL.

Comment (rouault) [reply gdal/port/cpl_conv.cpp]: The papszOptions argument is generally the last one in the parameter list in similar APIs

**Comment by rouault** [comment]
Apparently, nor Doxygen nor clang are happy with ```@param pnBufLength[out]```

**Comment by landam** [comment]
Any chance to include this PR in 2.3.0? Part of VFK bugfixes relies on CPLReadLine3L.

**Comment by rouault** [comment]
Feel free to cherry-pick the squashed commit in release/2.3

**Comment by landam** [comment]
For record: backport to release/2.3 done in dfac92801b

**Pull request merged**